### PR TITLE
Introducing a distance transform algorithm

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
One more piece of current work.

'dt_image_distance_transform' takes image data and does an eucledian distance transform.

The basic algorithm has been taken from a publication and reference implementation as:

*****
  Implementation of the distance transform algorithm described in:

  Distance Transforms of Sampled Functions
  Pedro F. Felzenszwalb and Daniel P. Huttenlocher
  Cornell Computing and Information Science TR2004-1963
  Copyright (C) 2006 Pedro Felzenszwalb
*****

the code has been published under GPL V2 or higher.

The original C++ code has been adopted to C and dt surrounding, omp support and less
pressure on alloc/dealloc has been tuned for performance in collaboration with Ingo Weyrich
from rawtherapee.

Code is running here already in some pending highlights reconstruction work, i would like to
get it reviewed / merged as it might be useful for other stuff too.

### CODE REVIEW

File: src/common/distance_transform.h [hunk 1]
```diff
@@ -0,0 +1,159 @@
+/*
+    This file is part of darktable,
+    Copyright (C) 2019-2021 darktable developers.
```
Comment (TurboGit) [inline src/common/distance_transform.h:3]: 2021 only.

File: src/iop/highlights.c [hunk 2]
```diff
@@ -24,6 +24,8 @@
 #include <string.h>
 #include "bauhaus/bauhaus.h"
 #include "common/opencl.h"
+#include "common/imagebuf.h"
```
Comment (TurboGit) [inline src/iop/highlights.c:27]: both line are superfluous at this point.

**Review by TurboGit — CHANGES_REQUESTED** [review]
First quick review. Some minor changes proposed. TIA.

File: src/common/distance_transform.h (hunk 1 above)
Comment (jenshannoschwalm) [reply src/common/distance_transform.h]: right

File: src/iop/highlights.c (hunk 2 above)
Comment (jenshannoschwalm) [reply src/iop/highlights.c]: right again. Forgot to remove after compile-testing.

**Review by TurboGit — APPROVED** [review]
Looks good, thanks !

**Pull request merged**